easybuild/easyblocks/b/bwise.py

```diff
@@ -0,0 +1,107 @@
+##
+# Copyright 2018 Free University of Brussels
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Ward Poelmans (Free University of Brussels)
+"""
+import glob
+import os
+import stat
+
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, copy_file, write_file
+
+
+class EB_BWISE(MakeCp):
+    """
+    Custom easyblock to install BWISE
+    """
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def build_step(self, verbose=False, path=None):
+        """Run multiple times for different sources"""
+
+        makefiles_fixes = [
+            (r'^CC=.*$', 'CC=$(CXX)'),
+            (r'^CFLAGS=.*$', 'CFLAGS:=$(CFLAGS)'),
+            (r'^LDFLAGS=.*$', 'LDFLAGS:=$(LDFLAGS) -fopenmp')
+        ]
+
+        os.chdir(self.builddir)
+        # first BWISE itself
+        bwisepath = glob.glob('BWISE-*')
+        if bwisepath:
+            os.chdir(os.path.join(bwisepath[0], 'src'))
+        else:
+            raise EasyBuildError("Could not find BWISE path")
+
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        apply_regex_substitutions('../Bwise.py', [(r'^BWISE_MAIN = .*$', 'BWISE_MAIN = os.environ[\'EBROOTBWISE\']')])
+        super(EB_BWISE, self).build_step()
+
+        # Onwards to BGREAT
+        os.chdir(self.builddir)
+        bgreatpath = glob.glob('BGREAT2-*')
+        if bgreatpath:
+            os.chdir(bgreatpath[0])
+        else:
+            raise EasyBuildError("Could not find BGREAT path")
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        super(EB_BWISE, self).build_step()
```

Comment:
This is way too similar to the above, please flesh this out in a helper method that you can call out to. Something like:

```python
# first BWISE itself
subdir = self.build_in_subdir(os.path.join('BWISE-*', 'src'))
bwise_py = os.path.join(subdir, '..', 'Bwise.py')
apply_regex_substitutions(bwise_py, [(r'^BWISE_MAIN = .*$', 'BWISE_MAIN = os.environ[\'EBROOTBWISE\']')])

# onwards to BGREAT
subdir = self.build_in_subdir('BGREAT-*')
copy_file(os.path.join(subdir, 'bgreat'), self.cfg['start_dir'])

# Finally, BTRIM
subdir = self.build_in_subdir('BTRIM-*')
copy_file(os.path.join(subdir, 'btrim'), self.cfg['start_dir'])